install/installer/scripts/kots-install.sh

```diff
@@ -0,0 +1,345 @@
+yq e -i ".domain = \"${DOMAIN}\"" "${CONFIG_FILE}"
+yq e -i '.license.kind = "secret"' "${CONFIG_FILE}"
+yq e -i '.license.name = "gitpod-license"' "${CONFIG_FILE}"
+
+if [ "${OPEN_VSX_URL}" != "" ];
+then
+    echo "Gitpod: Setting Open VSX Registry URL"
+    yq e -i ".openVSX.url = \"${OPEN_VSX_URL}\"" "${CONFIG_FILE}"
+fi
+
+if [ "${DB_INCLUSTER_ENABLED}" = "0" ] && [ "${DB_CLOUDSQL_INSTANCE}" != "" ];
+then
+    echo "Gitpod: configuring CloudSQLProxy"
+
+    yq e -i ".database.inCluster = false" "${CONFIG_FILE}"
+    yq e -i ".database.cloudSQL.instance = \"${DB_CLOUDSQL_INSTANCE}\"" "${CONFIG_FILE}"
+    yq e -i ".database.cloudSQL.serviceAccount.kind = \"secret\"" "${CONFIG_FILE}"
+    yq e -i ".database.cloudSQL.serviceAccount.name = \"cloudsql\"" "${CONFIG_FILE}"
+fi
+
+if [ "${DB_INCLUSTER_ENABLED}" = "0" ] && [ "${DB_CLOUDSQL_INSTANCE}" = "" ];
+then
+    echo "Gitpod: configuring external database"
+
+    yq e -i ".database.inCluster = false" "${CONFIG_FILE}"
+    yq e -i ".database.external.certificate.kind = \"secret\"" "${CONFIG_FILE}"
+    yq e -i ".database.external.certificate.name = \"database\"" "${CONFIG_FILE}"
+fi
+
+if [ "${HAS_LOCAL_REGISTRY}" = "true" ];
+then
+    echo "Gitpod: configuring mirrored container registry for airgapped installation"
+
+    yq e -i ".repository = \"${LOCAL_REGISTRY_ADDRESS}\"" "${CONFIG_FILE}"
+    yq e -i ".imagePullSecrets[0].kind = \"secret\"" "${CONFIG_FILE}"
+    yq e -i ".imagePullSecrets[0].name = \"${IMAGE_PULL_SECRET_NAME}\"" "${CONFIG_FILE}"
+    yq e -i '.dropImageRepo = true' "${CONFIG_FILE}"
+
+    # Add the registry to the server allowlist - keep docker.io in case it's just using the mirrored registry functionality without being airgapped
+    yq e -i ".containerRegistry.privateBaseImageAllowList += \"${LOCAL_REGISTRY_HOST}\"" "${CONFIG_FILE}"
+    yq e -i ".containerRegistry.privateBaseImageAllowList += \"docker.io\"" "${CONFIG_FILE}"
+fi
+
+if [ "${REG_CUSTOM_DOCKER_ENABLED}" = "1" ];
+then
+    echo "Gitpod: extracting servers from the custom registry authentication"
+
+    kubectl get secret \
+        -n "${NAMESPACE}" \
+        custom-registry-credentials \
+        -o jsonpath="{.data.\.dockerconfigjson}" | base64 -d > /tmp/userconfig.json
+
+    # Add the registries to the server allowlist
+    yq e -i ".containerRegistry.privateBaseImageAllowList += $(jq '.auths' /tmp/userconfig.json | jq -rc 'keys')" "${CONFIG_FILE}"
+    yq e -i ".containerRegistry.privateBaseImageAllowList += \"docker.io\"" "${CONFIG_FILE}"
+fi
+
+# Output the local registry secret - this is proxy.replicated.com if user hasn't set their own
+echo "${LOCAL_REGISTRY_IMAGE_PULL_SECRET}" | base64 -d > /tmp/kotsregistry.json
+
+if [ "${REG_INCLUSTER_ENABLED}" = "0" ];
+then
+    echo "Gitpod: configuring external container registry"
+
+    # Get the external-container-registry secret so we can merge the external registry and KOTS registry keys
+    kubectl get secret external-container-registry \
```

Comment:
:thumsup: Like that this is now routed through a secret, instead of building it here!